pkg/dynamiclifecycle/table.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamiclifecycle
+
+import (
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/index"
+	"github.com/cilium/statedb/reconciler"
+
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const TableName = "dynamic-lifecycles"
+
+var (
+	featureIndex = statedb.Index[TableLifecycle, DynamicFeature]{
+		Name: "feature",
+		FromObject: func(t TableLifecycle) index.KeySet {
+			return index.NewKeySet(index.Stringer(t.FeatureLifecycle.GetName()))
+		},
+		FromKey: index.Stringer[DynamicFeature],
+		Unique:  true,
+	}
+
+	ByFeature = featureIndex.Query
+
+	statusIndex = reconciler.NewStatusIndex(TableLifecycle.GetStatus)
+)
+
+type TableLifecycle struct {
```

Comment:
Weird name for this. Could this be just called `DynamicFeature` and what was `DynamicFeature` could be `DynamicFeatureName`.